`mapper_parsing_exception` when indexing queries

Creating a query set resulted in an error for the following configuration:

```
{
  "sampler": "all",
  "name": "",
  "description": "",
  "sampling": "",
  "querySetSize": 250
}
```
I suspect that something with the data being indexed either was incorrect or is handled incorrectly. 

Error message:
```
Search Quality Evaluation Framework
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
10:52:00.698 [main] INFO  org.opensearch.eval.samplers.AbstractQuerySampler - Indexing 47 queries for query set
Exception in thread "main" org.opensearch.client.opensearch._types.OpenSearchException: Request failed: [mapper_parsing_exception] failed to parse
	at org.opensearch.client.transport.httpclient5.ApacheHttpClient5Transport.extractAndWrapCause(ApacheHttpClient5Transport.java:1153)
	at org.opensearch.client.transport.httpclient5.ApacheHttpClient5Transport.performRequest(ApacheHttpClient5Transport.java:158)
	at org.opensearch.client.opensearch.OpenSearchClient.index(OpenSearchClient.java:874)
	at org.opensearch.eval.engine.OpenSearchEngine.indexQuerySet(OpenSearchEngine.java:147)
	at org.opensearch.eval.samplers.AbstractQuerySampler.indexQuerySet(AbstractQuerySampler.java:68)
	at org.opensearch.eval.samplers.AllQueriesQuerySampler.sample(AllQueriesQuerySampler.java:60)
	at org.opensearch.eval.App.main(App.java:120)
Caused by: org.opensearch.client.opensearch._types.OpenSearchException: Request failed: [mapper_parsing_exception] failed to parse
	at org.opensearch.client.transport.Endpoint.exceptionConverter(Endpoint.java:102)
	at org.opensearch.client.transport.httpclient5.ApacheHttpClient5Transport.prepareResponse(ApacheHttpClient5Transport.java:547)
	at org.opensearch.client.transport.httpclient5.ApacheHttpClient5Transport.lambda$performRequestAsync$0(ApacheHttpClient5Transport.java:191)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:646)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2179)
	at org.opensearch.client.transport.httpclient5.ApacheHttpClient5Transport$1.completed(ApacheHttpClient5Transport.java:236)
{
	at org.opensearch.client.transport.httpclient5.ApacheHttpClient5Transport$1.completed(ApacheHttpClient5Transport.java:225)
	at org.apache.hc.core5.concurrent.BasicFuture.completed(BasicFuture.java:148)
	at org.apache.hc.core5.concurrent.ComplexFuture.completed(ComplexFuture.java:72)
	at org.apache.hc.client5.http.impl.async.InternalAbstractHttpAsyncClient$2$1.completed(InternalAbstractHttpAsyncClient.java:310)
	at org.apache.hc.core5.http.nio.support.AbstractAsyncResponseConsumer$1.completed(AbstractAsyncResponseConsumer.java:101)
	at org.apache.hc.core5.http.nio.entity.AbstractBinAsyncEntityConsumer.completed(AbstractBinAsyncEntityConsumer.java:87)
	at org.apache.hc.core5.http.nio.entity.AbstractBinDataConsumer.streamEnd(AbstractBinDataConsumer.java:83)
	at org.apache.hc.core5.http.nio.support.AbstractAsyncResponseConsumer.streamEnd(AbstractAsyncResponseConsumer.java:142)
	at org.apache.hc.client5.http.impl.async.HttpAsyncMainClientExec$1.streamEnd(HttpAsyncMainClientExec.java:284)
	at org.apache.hc.core5.http.impl.nio.ClientHttp1StreamHandler.dataEnd(ClientHttp1StreamHandler.java:276)
	at org.apache.hc.core5.http.impl.nio.ClientHttp1StreamDuplexer.dataEnd(ClientHttp1StreamDuplexer.java:366)
	at org.apache.hc.core5.http.impl.nio.AbstractHttp1StreamDuplexer.onInput(AbstractHttp1StreamDuplexer.java:338)
	at org.apache.hc.core5.http.impl.nio.AbstractHttp1IOEventHandler.inputReady(AbstractHttp1IOEventHandler.java:64)
	at org.apache.hc.core5.http.impl.nio.ClientHttp1IOEventHandler.inputReady(ClientHttp1IOEventHandler.java:41)
	at org.apache.hc.core5.reactor.InternalDataChannel.onIOEvent(InternalDataChannel.java:143)
	at org.apache.hc.core5.reactor.InternalChannel.handleIOEvent(InternalChannel.java:51)
	at org.apache.hc.core5.reactor.SingleCoreIOReactor.processEvents(SingleCoreIOReactor.java:176)
	at org.apache.hc.core5.reactor.SingleCoreIOReactor.doExecute(SingleCoreIOReactor.java:125)
	at org.apache.hc.core5.reactor.AbstractSingleCoreIOReactor.execute(AbstractSingleCoreIOReactor.java:92)
	at org.apache.hc.core5.reactor.IOReactorWorker.run(IOReactorWorker.java:44)
	at java.base/java.lang.Thread.run(Thread.java:1583)
```

Additional information:
* Chorus OpenSearch
* 171 queries were in `ubi_queries` at the time the command was run
* Command: `docker compose run search_evaluation_framework java -jar /app/search-evaluation-framework.jar -o http://opensearch:9200 -s /app/files/create-query-set.json` (via the script `create-query-set.sh` in Chorus)
* Changing the number of queries to 2 works, so something with the existing queries in the ubi_queries index seems off or is not handled ideally

--- a/scripts/create-query-set.json
+++ b/scripts/create-query-set.json
@@ -3,5 +3,5 @@
   "name": "",
   "description": "",
   "sampling": "",
-  "querySetSize": 500
+  "querySetSize": 250
 }

--- a/src/main/java/org/opensearch/eval/App.java
+++ b/src/main/java/org/opensearch/eval/App.java
@@ -107,6 +107,8 @@ public class App {
 
             if(file.exists()) {
 
+                searchEngine.createQuerySetIndex();
+
                 final String jsonString = Files.readString(file.toPath(), StandardCharsets.UTF_8);
                 final JsonElement jsonElement = JsonParser.parseString(jsonString);
                 final JsonObject jsonObject = jsonElement.getAsJsonObject();

--- a/src/main/java/org/opensearch/eval/engine/OpenSearchEngine.java
+++ b/src/main/java/org/opensearch/eval/engine/OpenSearchEngine.java
@@ -916,7 +916,7 @@ public class OpenSearchEngine extends SearchEngine {
                     "                \"description\": { \"type\": \"text\" },\n" +
                     "                \"id\": { \"type\": \"keyword\" },\n" +
                     "                \"name\": { \"type\": \"keyword\" },\n" +
-                    "                \"querySetQueries\": { \"type\": \"object\" },\n" +
+                    "                \"query_set_queries\": { \"type\": \"object\" },\n" +
                     "                \"sampling\": { \"type\": \"keyword\" }\n" +
                     "              }\n" +
                     "          }";

--- a/src/main/java/org/opensearch/eval/engine/OpenSearchEngine.java
+++ b/src/main/java/org/opensearch/eval/engine/OpenSearchEngine.java
@@ -74,7 +74,6 @@ import java.util.Map;
 import java.util.Set;
 import java.util.UUID;
 
-import static org.opensearch.eval.Constants.DASHBOARD_METRICS_INDEX_NAME;
 import static org.opensearch.eval.Constants.QUERY_SETS_INDEX_NAME;
 import static org.opensearch.eval.judgments.clickmodel.coec.CoecClickModel.EVENT_CLICK;
 import static org.opensearch.eval.judgments.clickmodel.coec.CoecClickModel.EVENT_IMPRESSION;

--- a/src/main/java/org/opensearch/eval/engine/OpenSearchEngine.java
+++ b/src/main/java/org/opensearch/eval/engine/OpenSearchEngine.java
@@ -912,7 +912,7 @@ public class OpenSearchEngine extends SearchEngine {
 
             final String mapping = "{\n" +
                     "              \"properties\": {\n" +
-                    "                \"datetime\": { \"type\": \"date\", \"format\": \"strict_date_time\" },\n" +
+                    "                \"timestamp\": { \"type\": \"date\", \"format\": \"strict_date_time\" },\n" +
                     "                \"description\": { \"type\": \"text\" },\n" +
                     "                \"id\": { \"type\": \"keyword\" },\n" +
                     "                \"name\": { \"type\": \"keyword\" },\n" +

--- a/src/main/java/org/opensearch/eval/engine/OpenSearchEngine.java
+++ b/src/main/java/org/opensearch/eval/engine/OpenSearchEngine.java
@@ -74,6 +74,8 @@ import java.util.Map;
 import java.util.Set;
 import java.util.UUID;
 
+import static org.opensearch.eval.Constants.DASHBOARD_METRICS_INDEX_NAME;
+import static org.opensearch.eval.Constants.QUERY_SETS_INDEX_NAME;
 import static org.opensearch.eval.judgments.clickmodel.coec.CoecClickModel.EVENT_CLICK;
 import static org.opensearch.eval.judgments.clickmodel.coec.CoecClickModel.EVENT_IMPRESSION;
 import static org.opensearch.eval.judgments.clickmodel.coec.CoecClickModel.INDEX_QUERY_DOC_CTR;
@@ -898,6 +900,30 @@ public class OpenSearchEngine extends SearchEngine {
         }
 
         return judgmentsId;
+
+    }
+
+    @Override
+    public void createQuerySetIndex() throws Exception {
+
+        final boolean querySetsIndexExists = doesIndexExist(QUERY_SETS_INDEX_NAME);
+
+        if (!querySetsIndexExists) {
+
+            final String mapping = "{\n" +
+                    "              \"properties\": {\n" +
+                    "                \"datetime\": { \"type\": \"date\", \"format\": \"strict_date_time\" },\n" +
+                    "                \"description\": { \"type\": \"text\" },\n" +
+                    "                \"id\": { \"type\": \"keyword\" },\n" +
+                    "                \"name\": { \"type\": \"keyword\" },\n" +
+                    "                \"querySetQueries\": { \"type\": \"object\" },\n" +
+                    "                \"sampling\": { \"type\": \"keyword\" }\n" +
+                    "              }\n" +
+                    "          }";
+
+            createIndex(QUERY_SETS_INDEX_NAME, mapping);
+
+        }
 
     }
 

--- a/src/main/java/org/opensearch/eval/engine/SearchEngine.java
+++ b/src/main/java/org/opensearch/eval/engine/SearchEngine.java
@@ -17,6 +17,7 @@ public abstract class SearchEngine {
     public abstract boolean doesIndexExist(String index) throws IOException;
     public abstract boolean createIndex(String index, String mapping) throws IOException;
     public abstract boolean deleteIndex(String index) throws IOException;
+    public abstract void createQuerySetIndex() throws Exception;
 
     public abstract String getUserQuery(final String queryId) throws Exception;
     public abstract UbiQuery getQueryFromQueryId(final String queryId) throws Exception;

--- a/src/main/java/org/opensearch/eval/model/data/QuerySet.java
+++ b/src/main/java/org/opensearch/eval/model/data/QuerySet.java
@@ -8,16 +8,27 @@
  */
 package org.opensearch.eval.model.data;
 
+import com.fasterxml.jackson.annotation.JsonProperty;
+
 import java.util.Collection;
 import java.util.Map;
 import java.util.UUID;
 
 public class QuerySet extends AbstractData {
 
+    @JsonProperty("name")
     private String name;
+
+    @JsonProperty("description")
     private String description;
+
+    @JsonProperty("sampling")
     private String sampling;
+
+    @JsonProperty("query_set_queries")
     private Collection<Map<String, Long>> querySetQueries;
+
+    @JsonProperty("timestamp")
     private String timestamp;
 
     public QuerySet() {

